src/contracts/core/DelegationManager.sol

```diff
@@ -193,149 +208,99 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
+         */
+        if (_delegationApprover != address(0) && msg.sender != _delegationApprover && msg.sender != operator) {
+            // check the signature expiry
+            require(approverSignatureAndExpiry.expiry >= block.timestamp, "DelegationManager._delegate: approver signature expired");
+
+            // calculate the struct hash, then increment `delegationApprover`'s nonce
+            uint256 currentApproverNonce = delegationApproverNonce[_delegationApprover];
+            bytes32 approverStructHash = keccak256(
+                abi.encode(DELEGATION_APPROVAL_TYPEHASH, _delegationApprover, staker, operator, currentApproverNonce, approverSignatureAndExpiry.expiry)
+            );
+            unchecked {
+                delegationApproverNonce[_delegationApprover] = currentApproverNonce + 1;
+            }
+
+            // calculate the digest hash
+            bytes32 approverDigestHash = keccak256(abi.encodePacked("\x19\x01", domainSeparator(), approverStructHash));
 
-        // retrieve list of strategies and their shares from strategy manager
+            // actually check that the signature is valid
+            EIP1271SignatureUtils.checkSignature_EIP1271(_delegationApprover, approverDigestHash, approverSignatureAndExpiry.signature);
+        }
+
+        // retrieve `staker`'s list of strategies and the staker's shares in each strategy from the StrategyManager
         (IStrategy[] memory strategies, uint256[] memory shares) = strategyManager.getDeposits(staker);
 
-        // add strategy shares to delegate's shares
+        // add strategy shares to delegated `operator`'s shares
         uint256 stratsLength = strategies.length;
         for (uint256 i = 0; i < stratsLength;) {
-            // update the share amounts for each of the operator's strategies
+            // update the share amounts for each of the `operator`'s strategies
             operatorShares[operator][strategies[i]] += shares[i];
             unchecked {
                 ++i;
             }
         }
 
-        // call into hook in delegationTerms contract
-        _delegationReceivedHook(dt, staker, strategies, shares);
+        // record the delegation relation between the staker and operator, and emit an event
+        delegatedTo[staker] = operator;
+        emit StakerDelegated(staker, operator);
     }
 
     // VIEW FUNCTIONS
+    /**
+     * @notice Getter function for the current EIP-712 domain separator for this contract.
+     * @dev The domain separator will change in the event of a fork that changes the ChainID.
+     */
+    function domainSeparator() public view returns (bytes32) {
+        if (block.chainid == ORIGINAL_CHAIN_ID) {
+            return _DOMAIN_SEPARATOR;
+        }
+        else {
+            return _calculateDomainSeparator();
+        }
+    }
 
     /// @notice Returns 'true' if `staker` *is* actively delegated, and 'false' otherwise.
     function isDelegated(address staker) public view returns (bool) {
```

Comment:
isDelegated and isNotDelegated, should have only 1 internal function?

Reply:
Good point, I think we should just eliminate `isNotDelegated` entirely. IIRC there was initially some kind of weird, limbo-ish 3rd state but we moved away from that model long ago.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/97/commits/ccdfd071315668a7d1cb93706589bac167f3d8e4